scripts/setupTypeScriptRollup.js

```diff
@@ -0,0 +1,246 @@
+// @ts-check
+const fs = require('fs');
+const path = require('path');
+const { argv } = require('process');
+
+const projectRoot = argv[2] || path.join(__dirname, '..');
+
+console.log('Adding TypeScript with Rollup');
+
+// Add deps to pkg.json
+const pkgJSONPath = path.join(projectRoot, 'package.json');
+const packageJSON = JSON.parse(fs.readFileSync(pkgJSONPath, 'utf8'));
+packageJSON.devDependencies = Object.assign(packageJSON.devDependencies, {
+	'@rollup/plugin-typescript': '^6.0.0',
+	'@tsconfig/svelte': '^1.0.10',
+	'@types/compression': '^1.7.0',
+	'@types/node': '^14.11.1',
+	'@types/polka': '^0.5.1',
+	'svelte-check': '^1.0.46',
+	'svelte-preprocess': '^4.3.0',
+	'tslib': '^2.0.1',
+	'typescript': '^4.0.3'
+});
+
+// Add script for checking
+packageJSON.scripts = Object.assign(packageJSON.scripts, {
+	validate: 'svelte-check --ignore src/node_modules/@sapper'
+});
+
+// Write the package JSON
+fs.writeFileSync(pkgJSONPath, JSON.stringify(packageJSON, null, '  '));
+
+scanFolderAndReplace(path.join(projectRoot, 'src'));
+
+// replace js filenames to ts
+function scanFolderAndReplace(dir) {
+	const elements = fs.readdirSync(dir, { withFileTypes: true });
+
+	for (let i = 0; i < elements.length; i++) {
+		if (elements[i].isDirectory()) {
+			scanFolderAndReplace(path.join(dir, elements[i].name));
+		} else if (elements[i].name.match(/^((?!json).)*js$/)) {
+			fs.renameSync(
+				path.join(dir, elements[i].name),
+				path.join(dir, elements[i].name).replace('.js', '.ts')
+			);
+		}
+	}
+}
+
+// Switch the *.svelte file to use TS
+[
+	{
+		view: 'components/Nav',
+		vars: [{ name: 'segment', type: 'string' }]
+	},
+	{
+		view: 'routes/index'
+	},
+	{
+		view: 'routes/about'
+	},
+	{
+		view: 'routes/_layout',
+		vars: [{ name: 'segment', type: 'string' }]
+	},
+	{
+		view: 'routes/_error',
+		vars: [
+			{ name: 'status', type: 'number' },
+			{ name: 'error', type: 'Error' }
+		]
+	},
+	{
+		view: 'routes/blog/index',
+		vars: [
+			{ name: 'posts', type: '{ slug: string; title: string, html: any }[]' }
+		],
+		contextModule: [
+			{
+				js: '.then(r => r.json())',
+				ts: '.then((r: { json: () => any; }) => r.json())'
+			},
+			{
+				js: '.then(posts => {',
+				ts: '.then((posts: { slug: string; title: string, html: any }[]) => {'
+			}
+		]
+	},
+	{
+		view: 'routes/blog/[slug]',
+		vars: [{ name: 'post', type: '{ slug: string; title: string, html: any }' }]
+	}
+].forEach(({ view, vars, contextModule }) => {
+	const svelteFilePath = path.join(projectRoot, 'src', `${view}.svelte`);
+	let file = fs.readFileSync(svelteFilePath, 'utf8');
+
+	file = file.replace(/(?:<script)(( .*?)*?)>/gm, '<script$1 lang="ts">');
```

Comment:
If you run the script multiple times (e.g. because it crashes later) it will keep adding multiple `lang="ts"` attributes. It would be good to check if one already exists.

Reply:
The script purges itself after one time usage. It is further down in the file: https://github.com/sveltejs/sapper-template/pull/262/files#diff-9b1761910be238be8bde997d8219967bR220